specs/sharding/beacon-chain.md

```diff
@@ -497,67 +522,74 @@ def process_operations(state: BeaconState, body: BeaconBlockBody) -> None:
     for_ops(body.voluntary_exits, process_voluntary_exit)
 ```
 
-### New Attestation processing
-
-#### Updated `process_attestation`
+##### Extended Attestation processing
 
 ```python
 def process_attestation(state: BeaconState, attestation: Attestation) -> None:
     phase0.process_attestation(state, attestation)
-    update_pending_votes(state, attestation)
+    update_pending_shard_work(state, attestation)
 ```
 
-#### `update_pending_votes`
-
 ```python
-def update_pending_votes(state: BeaconState, attestation: Attestation) -> None:
-    # Find and update the PendingShardHeader object, invalid block if pending header not in state
-    if compute_epoch_at_slot(attestation.data.slot) == get_current_epoch(state):
-        pending_headers = state.current_epoch_pending_shard_headers
-    else:
-        pending_headers = state.previous_epoch_pending_shard_headers
-    
+def update_pending_shard_work(state: BeaconState, attestation: Attestation) -> None:
     attestation_shard = compute_shard_from_committee_index(
         state,
         attestation.data.slot,
         attestation.data.index,
     )
-    pending_header = None
-    for header in pending_headers:
-        if (
-            header.root == attestation.data.shard_header_root
-            and header.slot == attestation.data.slot
-            and header.shard == attestation_shard
-        ):
-            pending_header = header
-    assert pending_header is not None
-
-    for i in range(len(pending_header.votes)):
-        pending_header.votes[i] = pending_header.votes[i] or attestation.aggregation_bits[i]
-
-    # Check if the PendingShardHeader is eligible for expedited confirmation
-    # Requirement 1: nothing else confirmed
-    all_candidates = [
-        c for c in pending_headers if
-        (c.slot, c.shard) == (pending_header.slot, pending_header.shard)
-    ]
-    if True in [c.confirmed for c in all_candidates]:
+    buffer_index = attestation.data.slot % SHARD_STATE_MEMORY_SLOTS
+    committee_work = state.shard_buffer[buffer_index][attestation_shard]
+
+    # Skip attestation vote accounting if the header is not pending
+    if committee_work.status.selector != SHARD_WORK_PENDING:
+        # TODO In Altair: set participation bit flag, if attestation matches winning header.
```

Comment:
For the current version I believe we could not match `attestation.data.shard_header_root` since only `DataCommitment` remains (in case of `SHARD_WORK_CONFIRMED`)